main.tf

```diff
@@ -60,12 +60,15 @@ resource "aws_security_group" "default" {
   #checkov:skip=CKV2_AWS_5: "Ensure that Security Groups are attached to another resource" - False positive.
 
   description            = var.description
-  name                   = var.name
   name_prefix            = var.name_prefix
   revoke_rules_on_delete = var.revoke_rules_on_delete
-  tags                   = var.tags
   vpc_id                 = var.vpc_id
 
+  tags = merge(
+    var.tags,
+    { "NamePrefix" = var.name_prefix }
+  )
```

Comment:
I would keep this on one line

```
  tags = merge(var.tags, { "NamePrefix" = var.name_prefix })
```

Still looks clean to me. Any more, then yes let's move to own block.